SYS_ATL/LoopIR_scheduling.py

```diff
@@ -1203,6 +1203,46 @@ def map_e(self, e):
         return super().map_e(e)
 
 
+class _DoRearrangeDim(LoopIR_Rewrite):
+    def __init__(self, proc, alloc_stmt, dimensions):
+        assert isinstance(alloc_stmt, LoopIR.Alloc)
+
+        self.alloc_stmt = alloc_stmt
+        self.dimensions = dimensions
+
+        super().__init__(proc)
+
+        self.proc = InferEffects(self.proc).result()
+
+    def map_s(self, s):
+        # simply change the dimension
+        if s is self.alloc_stmt:
+            # construct new_hi
+            new_hi   = [s.type.hi[i] for i in self.dimensions]
+            # construct new_type
+            new_type = LoopIR.Tensor(new_hi, s.type.is_window, s.type.type)
+
+            return [LoopIR.Alloc(s.name, new_type, s.mem, None, s.srcinfo)]
```

Comment:
Imagine:
```
return [s.with(type=s.type.with(hi=[s.type.hi[i] for i in self.dimensions]))]
```
Or imagine adding a helper function to this class:
```
def _permute(self, xs):
    return [xs[i] for i in self.dimensions]
```
Then it could just be:
```
return [s.with(type=s.type.with(hi=self._permute(s.type.hi)))]
```